test/fail_compilation/dip25.d

```diff
@@ -2,7 +2,11 @@
 REQUIRED_ARGS: -de
 TEST_OUTPUT:
 ---
-fail_compilation/dip25.d(13): Deprecation: returning `this.buffer[]` escapes a reference to parameter `this`, perhaps annotate with `return`
+fail_compilation/dip25.d(17): Deprecation: returning `this.buffer[]` escapes a reference to parameter `this`
+fail_compilation/dip25.d(17):        perhaps annotate the parameter with `return`
+fail_compilation/dip25.d(22): Error: returning `identity(x)` escapes a reference to local variable `x`
```

Comment:
I think this error and error message is wrong:
* `x` is a parameter, not a local variable
* `x` is annotated with `return` so it should be safe to return even if passed through another `return` parameter.

Reply:
No it's not. That was the whole point of the issue, actually: `x` is of type `int`, so it's not a reference, hence why the message mention it's a local variable. And the `return` is useless, because of this.
We don't want to warn on useless `return` though, as that would make it harder to write generic code.

Reply:
Ugh nevermind me... I read the code at least 5 times as `fun(return ref int x)`....